src/main/java/kr/codesqaud/cafe/service/ReplyService.java

```diff
@@ -0,0 +1,46 @@
+package kr.codesqaud.cafe.service;
+
+import kr.codesqaud.cafe.domain.Reply;
+import kr.codesqaud.cafe.dto.ReplyDto;
+import kr.codesqaud.cafe.dto.ReplyForm;
+import kr.codesqaud.cafe.repository.JdbcReplyRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ReplyService {
+
+    private final JdbcReplyRepository replyRepository;
+
+    public ReplyService(JdbcReplyRepository replyRepository) {
+        this.replyRepository = replyRepository;
+    }
+
+    public void save(ReplyForm replyForm) {
+        Reply reply = Reply.builder()
+                .articleId(replyForm.getArticleId())
+                .writer(replyForm.getWriter())
+                .contents(replyForm.getContents())
+                .build();
+
+        replyRepository.save(reply);
+    }
+
+    public List<ReplyDto> findAllByArticleId(long articleId) {
+        List<Reply> replies = replyRepository.findAllByArticleId(articleId);
+        List<ReplyDto> replyDtos = new ArrayList<>();
+
+        for (Reply reply : replies) {
+            ReplyDto replyDto = ReplyDto.builder()
+                    .id(reply.getId())
+                    .writer(reply.getWriter())
+                    .contents(reply.getContents())
+                    .writtenTime(reply.getWrittenTime())
+                    .build();
+            replyDtos.add(replyDto);
+        }
+        return replyDtos;
+    }
```

Comment:
가져온 Reply 도메인 객체들을 DTO로 변경하는 정석적인 로직을 구현해주신 것 같습니다👍

p3)
그런데 여기서 조금 더 욕심을 부려보자면, 자바의 stream api를 이용하여 더 간결한 코드를 만들 수 있겠다는 생각이 듭니다.

```suggestion
    public List<ReplyDto> findAllByArticleId(long articleId) {
        return replyRepository.findAllByArticleId(articleId)
            .stream()
            .map(this::convertToDto)
            .collect(Collectors.toUnmodifiableList());
    }
    
    private ReplyDto convertToDto(Reply reply) {
        return ReplyDto.builder()
            .id(reply.getId())
            .writer(reply.getWriter())
            .contents(reply.getContents())
            .writtenTime(reply.getWrittenTime())
            .build();
    }
```

개인적으로 리뷰에 이렇게 코드를 작성해서 보여드리는건 좋은 리뷰가 아니라고 생각하는데요.
스트림을 이용하면 읽기에 간결한 코드를 작성할 수 있다는걸 보여드리기 위해 작성해봤습니다.

무조건 스트림을 사용해야한다는건 아니구요. 단지 스트림을 이용하면 어떤 형태의 코드를 작성할 수 있고
스트림을 사용할때와 사용하지 않을때 어떤 장단점들이 있는지 한번 생각해보시면 좋을 것 같다는 의견입니다.

장단점을 잘 생각해놓고 계시면 후에 코드를 작성하실때 둘 중에 선택할때 도움이 되실거라 생각합니다.